llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -1019,6 +1087,17 @@ static Error materializeCUDie(DWARFCompileUnit &DCU,
       memcpy(&SectionContents[SectionOffset],
              &DistinctDataArrayRef[DistinctDataOffset], *FormSize);
       DistinctDataOffset += *FormSize;
+    } else if (Form == llvm::dwarf::DW_FORM_sec_offset) {
+      memcpy(&SectionContents[SectionOffset], &SecOffsetVals[SecOffsetIndex],
+             *FormSize);
+      // If the size of the SecOffsetsVals vector is 1 then we only have one
+      // line table, so its offset is the only one we should use for all
+      // sec_offsets.
+      if (SecOffsetVals.size() > 1)
+        SecOffsetIndex++;
+      else if (SecOffsetVals.size() == 0)
+        llvm_unreachable(
+            "SecOffset SmallVector should have a non-zero length!");
```

Comment:
If we can express the same intent without control flow, it's easier to read:

```cpp
assert(SecOffsetVals.size() != 0 && "SecOffset SmallVector should have a non-zero length!"))
```

Reply:
That is fair!